### DISCUSSION TITLE
Enhancements on backup / document_exporter

### DISCUSSION THREAD

**Discussion by karabays** [jonaswinkler/paperless-ng#135] [Archived FRs]
Hi, really appreciate your work here!

I was wondering if it is possible to enhance the backup feature little bit. Seems document_exporter works fine (at least exporting nicely, didn't try importing yet...) but it would be more useful if exports can be scheduled and create a nice & tidy tar file with timestamp. so it's easier to move the backup to an external drive or something.

**Comment by jonaswinkler** [comment]
Hello, and thanks for your suggestion!

I'm doing backups with the exporter and `rsync` right now. Most, if not all of the files in paperless don't really ever change. With `rsync --link-dest`, rsync only copies changed and added files, whereas unchanged files are hard linked to previous backups. Keeps the backup size down. That's impossible when the entire folder is in a compressed tar archive.

You could also try scheduling a shell script instead of the export command directly, which would export to a temporary folder, create a compressed archive there and move that into the destination folder.

I don't know if this should be something that paperless should support natively when its fairly easy to add that with some external tools.

**Reply by karabays** [reply]
incremental backup is a valid point I guess. I am using duplicati but running docker-compose with sudo and emptying export folder every time felt little bit cumbersome.

**Comment by jcpsantiago** [comment]
I also think this should be done by an external tool/script (at least for now), instead of adding another features to paperless. I'll try giving it a try.
I'm testing Arq for my backups -- it supports pre- and post- scripts which could stop paperless, export docs, cleanup and restart. The same is possible with `rsync`, `borg` or `rcloud` or `<insert cli tool>` in a shell script. 

Anyway I'll put something together and share it.

**Comment by vistalba** [comment]
Is it possible to schedule document_exporter with the scheduled tasks feature? Didn't get it work :/
